auth/api/iam/api.go

```diff
@@ -153,3 +156,29 @@ func (r Wrapper) HandleAuthorizeRequest(ctx context.Context, request HandleAutho
 		//Description: "missing or invalid parameters",
 	}
 }
+
+// GetOAuthAuthorizationServerMetadata returns the Authorization Server's metadata
+func (r Wrapper) GetOAuthAuthorizationServerMetadata(ctx context.Context, request GetOAuthAuthorizationServerMetadataRequestObject) (GetOAuthAuthorizationServerMetadataResponseObject, error) {
+	id, err := did.ParseDID(request.Did)
+	if err != nil {
+		return nil, err
+	}
+
+	if id.Method != "nuts" {
+		return nil, errors.New("only did:nuts is supported")
+	}
+
+	owned, err := r.vdr.IsOwner(ctx, *id)
+	if err != nil {
+		// TODO: should this be not found?
```

Comment:
yes. Maybe in combination with `IsFunctionalError`; DID resolution errors like NotFound, Deactivated etc can lead to 404, other technical errors (database error) could lead to 500 (and need to be logged)